app/src/main/java/io/gnosis/safe/ui/settings/app/SettingsHandler.kt

```diff
@@ -16,8 +20,51 @@ import javax.inject.Singleton
 @Singleton
 class SettingsHandler @Inject constructor(
     private val gatewayApi: GatewayApi,
-    private val preferencesManager: PreferencesManager
+    private val preferencesManager: PreferencesManager,
+    private val remoteConfig: FirebaseRemoteConfig
 ) {
+
+    fun fetchRemoteConfig() {
+        remoteConfig.fetchAndActivate().addOnCompleteListener {
+            if (it.isSuccessful) {
+               updateUpdateInfo()
+            }
+        }
+    }
+
+    var updateNewVersionShown: Boolean
+        get() {
+            val version = preferencesManager.prefs.getInt(KEY_UPDATE_SHOWN_FOR_VERSION, -1)
+            // show update dialog for every new version if needed
+            return version == BuildConfig.VERSION_CODE
+        }
+        set(value) {
+            preferencesManager.prefs.edit {
+                putInt(KEY_UPDATE_SHOWN_FOR_VERSION, if (value) BuildConfig.VERSION_CODE else -1)
+            }
+        }
+
+    var updateNewestVersion: Boolean = false
+        private set
+    var updateDeprecatedSoon: Boolean = false
+        private set
+    var updateDeprecated: Boolean = false
+        private set
+
+    val showUpdateInfo: Boolean =
+        updateNewVersionShown && !updateNewVersionShown || updateDeprecatedSoon || updateDeprecated
```

Comment:
Apparently `||` has precedence over `&&` (Otherwise `updateNewVersionShown && !updateNewVersionShown` wouldn't make much sense. But to make this easier to comprehend: Could you add some clarifying parentheses?